farmdata2_modules/fd2_tabs/resources/RegexInputComponent.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * A Vue component for input HTML elements that use Regex to validate user input
+ * 
+ * <p><b>data-cy attributes</b></p>
+ * <table>
+ * <thead><tr><th>Value</th>        <th>Descripion</th></tr></thead>
+ * <tbody>
+ * <tr><td>dropdown-component</td>  <td>div containing the label and select elements.</td></tr>
+ * <tr><td>dropdown-input</td>      <td>The select element.</td></tr>
+ * <tr><td>optioni</td>             <td>The ith option element, i=0,1,2...</td></tr>
+ * </tbody>
+ * </table>
+ * 
+ * @vue-prop {RegExp} [regExp] - The regular expression used to validate user inputs.
+ * @vue-prop {String} [defaultVal] - Used to clear the page once a log has been submitted.
+ * @vue-prop {String} [setType] - Defines the type of input box.
+ * @vue-prop {String} [setColor] - Defines the background color of the input element.
+ * @vue-prop {String} [setHeight] - Defines the height of the input element.
+ * @vue-prop {String} [setWidth] - Defines the width of the input element.
+ * @vue-prop {String} [setMin] - Defines the min of the input element.
+ * @vue-prop {String} [setMax] - Defines the max of the input element.
+ * @vue-prop {String} [setStep] - Defines the step increments of the input element.
+ * @vue-prop {String} [setTitle] - Defines the title for a simple hover tip for the input element.
+ * @vue-prop {Boolean} [isDisabled] - Makes the element read-only.
+ * 
+ * @vue-event {Boolean} is-enabled - Emits boolean (isMatch) for the parent page to make use of the results of the validation. 
```

Comment:
It looks like you have two events emitted in the code.  This documentation should match what the code is doing.